src/ray/core_worker/core_worker.cc

```diff
@@ -108,6 +119,48 @@ CoreWorker::CoreWorker(
   }
 }
 
+void CoreWorker::AddActiveObjectID(const ObjectID &object_id) {
+  io_service_.post([this, object_id]() -> void {
+    active_object_ids_.insert(object_id);
+    active_object_ids_updated_ = true;
+  });
+}
+
+void CoreWorker::RemoveActiveObjectID(const ObjectID &object_id) {
+  io_service_.post([this, object_id]() -> void {
+    if (active_object_ids_.erase(object_id)) {
+      active_object_ids_updated_ = true;
+    } else {
+      RAY_LOG(WARNING) << "Tried to erase non-existent object ID" << object_id;
+    }
+  });
+}
+
+void CoreWorker::SendActiveObjectIDsHeartbeat() {
+  // Only send a heartbeat when the set of active object IDs has changed because the
+  // raylet only modifies the set of IDs when it receives a heartbeat.
+  if (active_object_ids_updated_) {
+    std::cout << "Sending " << active_object_ids_.size() << " object IDs to raylet."
+              << std::endl;
+    if (active_object_ids_.size() >
+        static_cast<size_t>(RayConfig::instance().raylet_active_object_ids_size())) {
```

Comment:
Maybe just define `raylet_active_object_ids_size` to be a `size_t` in the config.